### ISSUE TITLE
Generated composer.lock file is empty

### ISSUE THREAD

**Issue by Terimor** [composer/composer#11800]
Hello guys,

I got a problem with package management.

My `composer.json`:

```
{
  "name": "",
  "version": "1.0.0",
  "license": "proprietary",
  "type": "project",
  "repository": "",
  "authors": [
    {
      "name": "",
      "email": ""
    }
  ],
  "require": {
    "php": ">=8.3",
    "ext-dom": "*",
    "ext-iconv": "*",
    "ext-json": "*",
    "ext-pdo": "*",
    "ext-redis": "*",
    "ext-soap": "*",
    "ext-ssh2": "*",
    "ext-zip": "*",
    "doctrine/annotations": "^1.10",
    "doctrine/doctrine-bundle": "^2.2",
    "doctrine/doctrine-migrations-bundle": "^3.1",
    "doctrine/orm": "^2.10",
    "enqueue/amqp-lib": "^0.10.19",
    "enqueue/enqueue-bundle": "^0.10.15",
    "enqueue/redis": "^0.10.19",
    "enqueue/simple-client": "0.10.9",
    "friendsofphp/php-cs-fixer": "^2.17",
    "friendsofsymfony/elastica-bundle": "^6.0",
    "guzzlehttp/guzzle": "6.5.5",
    "jms/serializer-bundle": "^3.8.0",
    "mpdf/mpdf": "^8.2",
    "php-webdriver/webdriver": "^1.14",
    "phpoffice/phpspreadsheet": "^1.17",
    "phpseclib/phpseclib": "3.0.34",
    "phpunit/phpunit": "^10",
    "predis/predis": "^1.1",
    "snc/redis-bundle": "^3.3",
    "symfony/asset": "^5.4",
    "symfony/console": "^5.4",
    "symfony/event-dispatcher": "^5.4",
    "symfony/flex": "^1.19",
    "symfony/form": "^5.4",
    "symfony/framework-bundle": "^5.4",
    "symfony/ldap": "^5.4",
    "symfony/lock": "^5.4",
    "symfony/security-bundle": "^5.4",
    "symfony/security-csrf": "^5.4",
    "symfony/stopwatch": "^5.4",
    "symfony/twig-bundle": "^5.4",
    "symfony/validator": "^6.0"
  },
  "require-dev": {
    "symfony/dotenv": "^5.4",
    "symfony/maker-bundle": "1.44.0",
    "symfony/phpunit-bridge": "^5.4",
    "symfony/web-profiler-bundle": "^5.4",
    "verdant/xml2array": "*@dev",
    "wsdltophp/wssecurity": "1.2.2"
  },
  "config": {
    "preferred-install": {
      "*": "dist"
    },
    "sort-packages": true,
    "allow-plugins": {
      "ocramius/package-versions": true,
      "symfony/flex": true
    }
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "App\\Tests\\": "tests/"
    }
  },
  "replace": {
    "symfony/polyfill-iconv": "*",
    "symfony/polyfill-php71": "*",
    "symfony/polyfill-php70": "*",
    "symfony/polyfill-php56": "*"
  },
  "scripts": {
    "auto-scripts": {
      "cache:clear": "symfony-cmd",
      "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd",
      "assets:install %PUBLIC_DIR%": "symfony-cmd"
    },
    "post-install-cmd": [
      "@auto-scripts"
    ],
    "post-update-cmd": [
      "@auto-scripts"
    ]
  },
  "conflict": {
    "symfony/symfony": "*"
  },
  "extra": {
    "symfony": {
      "allow-contrib": false,
      "id": "01DJWXBAY97SESJGWE44BC88Y1"
    }
  },
  "minimum-stability": "stable",
  "prefer-stable": true,
  "repositories": {
    "packagist": {
      "type": "composer",
      "url": "https://packagist.org"
    }
  }
}
```

Output of `composer diagnose`:
```
PHP Deprecated:  PHP Startup: Use of mbstring.internal_encoding is deprecated in Unknown on line 0
Checking composer.json: WARNING
The version field is present, it is recommended to leave it out if the package is published on Packagist.
require.guzzlehttp/guzzle : exact version constraints (6.5.5) should be avoided if the package follows semantic versioning
require.phpseclib/phpseclib : exact version constraints (3.0.34) should be avoided if the package follows semantic versioning
Checking platform settings: OK
Checking git settings: OK git version 2.39.2
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: You are not running the latest snapshot version, run `composer self-update` to update (2.6.6 => 4e5be9ee7d924d8efc58d676439b0c7bd18a9ce4)
Composer version: 2.6.6
PHP version: 8.3.0
PHP binary path: /usr/local/bin/php
OpenSSL version: OpenSSL 3.0.11 19 Sep 2023
cURL version: 7.88.1 libz 1.2.13 ssl OpenSSL/3.0.11
zip: extension present, unzip present, 7-Zip not available
```

When I run this command: 

```
composer update -vvv
```

I get the following output: 

```
(top is truncated due to github limitations)
Built pool.
Running pool optimizer.
Pool optimizer completed in 0.346 seconds
Found 6,521 package versions referenced in your dependency graph. 5,186 (80%) were optimized away.
Updating dependencies
Generating rules
Resolving dependencies through SAT
Looking at all rules.
Something's changed, looking at all rules again (pass #7)
Dependency resolution completed in 0.071 seconds
Analyzed 1335 packages to resolve dependencies
Analyzed 14231 rules to resolve dependencies
Generating rules
Resolving dependencies through SAT
Looking at all rules.

Dependency resolution completed in 0.005 seconds
Lock file operations: 159 installs, 0 updates, 0 removals
Installs: symfony/flex:v1.21.4, composer/pcre:1.0.1, symfony/var-exporter:v7.0.2, symfony/deprecation-contracts:v3.4.0, psr/container:1.1.2, symfony/service-contracts:v2.5.2, symfony/stopwatch:v5.4.21, symfony/polyfill-mbstring:v1.28.0, symfony/polyfill-intl-normalizer:v1.28.0, symfony/polyfill-intl-grapheme:v1.28.0, symfony/polyfill-ctype:v1.28.0, symfony/string:v6.4.2, symfony/polyfill-php80:v1.28.0, symfony/polyfill-php73:v1.28.0, symfony/console:v5.4.34, psr/log:2.0.0, doctrine/deprecations:1.1.2, doctrine/event-manager:1.2.0, psr/cache:3.0.0, doctrine/cache:2.2.0, doctrine/dbal:3.7.2, doctrine/migrations:3.7.2, symfony/routing:v6.1.11, symfony/finder:v5.4.27, symfony/filesystem:v5.4.25, symfony/polyfill-php81:v1.28.0, symfony/polyfill-php83:v1.28.0, symfony/http-foundation:v6.4.2, psr/event-dispatcher:1.0.0, symfony/event-dispatcher-contracts:v3.4.0, symfony/event-dispatcher:v5.4.34, symfony/var-dumper:v6.4.2, symfony/error-handler:v6.3.5, symfony/http-kernel:v6.0.20, symfony/dependency-injection:v5.4.34, symfony/config:v5.4.31, symfony/cache-contracts:v3.4.0, symfony/cache:v6.4.2, symfony/framework-bundle:v5.4.34, doctrine/persistence:3.2.0, symfony/doctrine-bridge:v5.4.34, doctrine/sql-formatter:1.1.3, doctrine/doctrine-bundle:2.11.1, doctrine/doctrine-migrations-bundle:3.3.0, symfony/polyfill-php72:v1.28.0, doctrine/lexer:2.1.0, doctrine/instantiator:2.0.0, doctrine/inflector:2.0.9, doctrine/common:3.4.3, doctrine/collections:2.1.4, doctrine/orm:2.17.3, enqueue/dsn:0.10.19, queue-interop/queue-interop:0.8.1, queue-interop/amqp-interop:0.8.2, enqueue/amqp-tools:0.10.19, paragonie/random_compat:v9.99.100, paragonie/constant_time_encoding:v2.6.3, phpseclib/phpseclib:3.0.34, php-amqplib/php-amqplib:v3.6.0, enqueue/amqp-lib:0.10.19, enqueue/null:0.10.19, ramsey/collection:2.0.0, brick/math:0.11.0, ramsey/uuid:4.7.5, enqueue/enqueue:0.10.20, enqueue/enqueue-bundle:0.10.19, enqueue/redis:0.10.19, enqueue/simple-client:0.10.9, react/promise:v2.11.0, ezimuel/guzzlestreams:3.1.0, ezimuel/ringphp:1.2.2, symfony/process:v5.4.34, symfony/options-resolver:v5.4.21, php-cs-fixer/diff:v1.3.1, doctrine/annotations:1.14.3, composer/xdebug-handler:2.0.5, composer/semver:3.4.0, friendsofphp/php-cs-fixer:v2.19.3, symfony/property-info:v6.4.0, symfony/property-access:v6.4.0, nyholm/dsn:2.0.1, elasticsearch/elasticsearch:v7.17.2, ruflin/elastica:7.3.1, pagerfanta/pagerfanta:v3.8.0, friendsofsymfony/elastica-bundle:v6.3.1, ralouphie/getallheaders:3.0.3, psr/http-message:1.1, guzzlehttp/psr7:1.9.1, guzzlehttp/promises:1.5.3, symfony/polyfill-intl-idn:v1.28.0, guzzlehttp/guzzle:6.5.5, jms/metadata:2.8.0, phpstan/phpdoc-parser:1.25.0, jms/serializer:3.29.1, jms/serializer-bundle:3.10.0, setasign/fpdi:v2.6.0, myclabs/deep-copy:1.11.1, mpdf/psr-log-aware-trait:v2.0.0, mpdf/psr-http-message-shim:1.0.0, mpdf/mpdf:v8.2.2, php-webdriver/webdriver:1.15.1, psr/simple-cache:3.0.0, psr/http-factory:1.0.2, psr/http-client:1.0.3, markbaker/matrix:3.0.1, markbaker/complex:3.0.2, maennchen/zipstream-php:3.1.0, ezyang/htmlpurifier:v4.17.0, phpoffice/phpspreadsheet:1.29.0, sebastian/version:4.0.1, sebastian/type:4.0.0, sebastian/recursion-context:5.0.0, sebastian/object-reflector:3.0.0, sebastian/object-enumerator:5.0.0, sebastian/global-state:6.0.1, sebastian/exporter:5.1.1, sebastian/environment:6.0.1, sebastian/diff:5.1.0, sebastian/comparator:5.0.1, sebastian/code-unit:2.0.0, sebastian/cli-parser:2.0.0, phpunit/php-timer:6.0.0, phpunit/php-text-template:3.0.1, phpunit/php-invoker:4.0.0, phpunit/php-file-iterator:4.1.0, theseer/tokenizer:1.2.2, nikic/php-parser:v4.18.0, sebastian/lines-of-code:2.0.2, sebastian/complexity:3.2.0, sebastian/code-unit-reverse-lookup:3.0.0, phpunit/php-code-coverage:10.1.11, phar-io/version:3.2.1, phar-io/manifest:2.0.3, phpunit/phpunit:10.5.7, predis/predis:v1.1.10, symfony/yaml:v6.4.0, snc/redis-bundle:3.6.0, symfony/asset:v5.4.31, symfony/dotenv:v5.4.34, symfony/polyfill-intl-icu:v1.28.0, symfony/form:v5.4.33, symfony/ldap:v5.4.27, symfony/lock:v5.4.34, symfony/maker-bundle:v1.44.0, symfony/phpunit-bridge:v5.4.34, symfony/password-hasher:v6.4.0, symfony/security-core:v5.4.30, symfony/security-http:v5.4.31, symfony/security-guard:v5.4.27, symfony/security-csrf:v5.4.27, symfony/security-bundle:v5.4.34, twig/twig:v3.8.0, symfony/translation-contracts:v3.4.1, symfony/twig-bridge:v5.4.34, symfony/validator:v6.4.2, symfony/twig-bundle:v5.4.31, symfony/web-profiler-bundle:v5.4.34, verdant/xml2array:dev-master 8afd54f, wsdltophp/wssecurity:1.2.2
  - Locking brick/math (0.11.0)
  - Locking composer/pcre (1.0.1)
  - Locking composer/semver (3.4.0)
  - Locking composer/xdebug-handler (2.0.5)
  - Locking doctrine/annotations (1.14.3)
  - Locking doctrine/cache (2.2.0)
  - Locking doctrine/collections (2.1.4)
  - Locking doctrine/common (3.4.3)
  - Locking doctrine/dbal (3.7.2)
  - Locking doctrine/deprecations (1.1.2)
  - Locking doctrine/doctrine-bundle (2.11.1)
  - Locking doctrine/doctrine-migrations-bundle (3.3.0)
  - Locking doctrine/event-manager (1.2.0)
  - Locking doctrine/inflector (2.0.9)
  - Locking doctrine/instantiator (2.0.0)
  - Locking doctrine/lexer (2.1.0)
  - Locking doctrine/migrations (3.7.2)
  - Locking doctrine/orm (2.17.3)
  - Locking doctrine/persistence (3.2.0)
  - Locking doctrine/sql-formatter (1.1.3)
  - Locking elasticsearch/elasticsearch (v7.17.2)
  - Locking enqueue/amqp-lib (0.10.19)
  - Locking enqueue/amqp-tools (0.10.19)
  - Locking enqueue/dsn (0.10.19)
  - Locking enqueue/enqueue (0.10.20)
  - Locking enqueue/enqueue-bundle (0.10.19)
  - Locking enqueue/null (0.10.19)
  - Locking enqueue/redis (0.10.19)
  - Locking enqueue/simple-client (0.10.9)
  - Locking ezimuel/guzzlestreams (3.1.0)
  - Locking ezimuel/ringphp (1.2.2)
  - Locking ezyang/htmlpurifier (v4.17.0)
  - Locking friendsofphp/php-cs-fixer (v2.19.3)
  - Locking friendsofsymfony/elastica-bundle (v6.3.1)
  - Locking guzzlehttp/guzzle (6.5.5)
  - Locking guzzlehttp/promises (1.5.3)
  - Locking guzzlehttp/psr7 (1.9.1)
  - Locking jms/metadata (2.8.0)
  - Locking jms/serializer (3.29.1)
  - Locking jms/serializer-bundle (3.10.0)
  - Locking maennchen/zipstream-php (3.1.0)
  - Locking markbaker/complex (3.0.2)
  - Locking markbaker/matrix (3.0.1)
  - Locking mpdf/mpdf (v8.2.2)
  - Locking mpdf/psr-http-message-shim (1.0.0)
  - Locking mpdf/psr-log-aware-trait (v2.0.0)
  - Locking myclabs/deep-copy (1.11.1)
  - Locking nikic/php-parser (v4.18.0)
  - Locking nyholm/dsn (2.0.1)
  - Locking pagerfanta/pagerfanta (v3.8.0)
  - Locking paragonie/constant_time_encoding (v2.6.3)
  - Locking paragonie/random_compat (v9.99.100)
  - Locking phar-io/manifest (2.0.3)
  - Locking phar-io/version (3.2.1)
  - Locking php-amqplib/php-amqplib (v3.6.0)
  - Locking php-cs-fixer/diff (v1.3.1)
  - Locking php-webdriver/webdriver (1.15.1)
  - Locking phpoffice/phpspreadsheet (1.29.0)
  - Locking phpseclib/phpseclib (3.0.34)
  - Locking phpstan/phpdoc-parser (1.25.0)
  - Locking phpunit/php-code-coverage (10.1.11)
  - Locking phpunit/php-file-iterator (4.1.0)
  - Locking phpunit/php-invoker (4.0.0)
  - Locking phpunit/php-text-template (3.0.1)
  - Locking phpunit/php-timer (6.0.0)
  - Locking phpunit/phpunit (10.5.7)
  - Locking predis/predis (v1.1.10)
  - Locking psr/cache (3.0.0)
  - Locking psr/container (1.1.2)
  - Locking psr/event-dispatcher (1.0.0)
  - Locking psr/http-client (1.0.3)
  - Locking psr/http-factory (1.0.2)
  - Locking psr/http-message (1.1)
  - Locking psr/log (2.0.0)
  - Locking psr/simple-cache (3.0.0)
  - Locking queue-interop/amqp-interop (0.8.2)
  - Locking queue-interop/queue-interop (0.8.1)
  - Locking ralouphie/getallheaders (3.0.3)
  - Locking ramsey/collection (2.0.0)
  - Locking ramsey/uuid (4.7.5)
  - Locking react/promise (v2.11.0)
  - Locking ruflin/elastica (7.3.1)
  - Locking sebastian/cli-parser (2.0.0)
  - Locking sebastian/code-unit (2.0.0)
  - Locking sebastian/code-unit-reverse-lookup (3.0.0)
  - Locking sebastian/comparator (5.0.1)
  - Locking sebastian/complexity (3.2.0)
  - Locking sebastian/diff (5.1.0)
  - Locking sebastian/environment (6.0.1)
  - Locking sebastian/exporter (5.1.1)
  - Locking sebastian/global-state (6.0.1)
  - Locking sebastian/lines-of-code (2.0.2)
  - Locking sebastian/object-enumerator (5.0.0)
  - Locking sebastian/object-reflector (3.0.0)
  - Locking sebastian/recursion-context (5.0.0)
  - Locking sebastian/type (4.0.0)
  - Locking sebastian/version (4.0.1)
  - Locking setasign/fpdi (v2.6.0)
  - Locking snc/redis-bundle (3.6.0)
  - Locking symfony/asset (v5.4.31)
  - Locking symfony/cache (v6.4.2)
  - Locking symfony/cache-contracts (v3.4.0)
  - Locking symfony/config (v5.4.31)
  - Locking symfony/console (v5.4.34)
  - Locking symfony/dependency-injection (v5.4.34)
  - Locking symfony/deprecation-contracts (v3.4.0)
  - Locking symfony/doctrine-bridge (v5.4.34)
  - Locking symfony/dotenv (v5.4.34)
  - Locking symfony/error-handler (v6.3.5)
  - Locking symfony/event-dispatcher (v5.4.34)
  - Locking symfony/event-dispatcher-contracts (v3.4.0)
  - Locking symfony/filesystem (v5.4.25)
  - Locking symfony/finder (v5.4.27)
  - Locking symfony/flex (v1.21.4)
  - Locking symfony/form (v5.4.33)
  - Locking symfony/framework-bundle (v5.4.34)
  - Locking symfony/http-foundation (v6.4.2)
  - Locking symfony/http-kernel (v6.0.20)
  - Locking symfony/ldap (v5.4.27)
  - Locking symfony/lock (v5.4.34)
  - Locking symfony/maker-bundle (v1.44.0)
  - Locking symfony/options-resolver (v5.4.21)
  - Locking symfony/password-hasher (v6.4.0)
  - Locking symfony/phpunit-bridge (v5.4.34)
  - Locking symfony/polyfill-ctype (v1.28.0)
  - Locking symfony/polyfill-intl-grapheme (v1.28.0)
  - Locking symfony/polyfill-intl-icu (v1.28.0)
  - Locking symfony/polyfill-intl-idn (v1.28.0)
  - Locking symfony/polyfill-intl-normalizer (v1.28.0)
  - Locking symfony/polyfill-mbstring (v1.28.0)
  - Locking symfony/polyfill-php72 (v1.28.0)
  - Locking symfony/polyfill-php73 (v1.28.0)
  - Locking symfony/polyfill-php80 (v1.28.0)
  - Locking symfony/polyfill-php81 (v1.28.0)
  - Locking symfony/polyfill-php83 (v1.28.0)
  - Locking symfony/process (v5.4.34)
  - Locking symfony/property-access (v6.4.0)
  - Locking symfony/property-info (v6.4.0)
  - Locking symfony/routing (v6.1.11)
  - Locking symfony/security-bundle (v5.4.34)
  - Locking symfony/security-core (v5.4.30)
  - Locking symfony/security-csrf (v5.4.27)
  - Locking symfony/security-guard (v5.4.27)
  - Locking symfony/security-http (v5.4.31)
  - Locking symfony/service-contracts (v2.5.2)
  - Locking symfony/stopwatch (v5.4.21)
  - Locking symfony/string (v6.4.2)
  - Locking symfony/translation-contracts (v3.4.1)
  - Locking symfony/twig-bridge (v5.4.34)
  - Locking symfony/twig-bundle (v5.4.31)
  - Locking symfony/validator (v6.4.2)
  - Locking symfony/var-dumper (v6.4.2)
  - Locking symfony/var-exporter (v7.0.2)
  - Locking symfony/web-profiler-bundle (v5.4.34)
  - Locking symfony/yaml (v6.4.0)
  - Locking theseer/tokenizer (1.2.2)
  - Locking twig/twig (v3.8.0)
  - Locking verdant/xml2array (dev-master 8afd54f)
  - Locking wsdltophp/wssecurity (1.2.2)
Reading ./composer.lock (/var/www/cmdb.localhost/composer.lock)
Writing lock file
Installing dependencies from lock file (including require-dev)
Reading ./composer.lock (/var/www/cmdb.localhost/composer.lock)

In JsonFile.php line 375:
                                                                          
  [Seld\JsonLint\ParsingException]                                        
  "./composer.lock" does not contain valid JSON                           
  Parse error on line 1:                                                  
                                                                          
  ^                                                                       
  Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['  
                                                                          

Exception trace:
  at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Json/JsonFile.php:375
 Composer\Json\JsonFile::validateSyntax() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Json/JsonFile.php:341
 Composer\Json\JsonFile::parseJson() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Json/JsonFile.php:126
 Composer\Json\JsonFile->read() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Package/Locker.php:330
 Composer\Package\Locker->getLockData() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Package/Locker.php:156
 Composer\Package\Locker->getLockedRepository() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Installer.php:700
 Composer\Installer->doInstall() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Installer.php:629
 Composer\Installer->doUpdate() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Installer.php:290
 Composer\Installer->run() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Command/UpdateCommand.php:249
 Composer\Command\UpdateCommand->execute() at phar:///var/www/cmdb.localhost/composer.phar/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///var/www/cmdb.localhost/composer.phar/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at phar:///var/www/cmdb.localhost/composer.phar/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Console/Application.php:382
 Composer\Console\Application->doRun() at phar:///var/www/cmdb.localhost/composer.phar/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///var/www/cmdb.localhost/composer.phar/src/Composer/Console/Application.php:145
 Composer\Console\Application->run() at phar:///var/www/cmdb.localhost/composer.phar/bin/composer:93
 require() at /var/www/cmdb.localhost/composer.phar:29

update [--with WITH] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--no-autoloader] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>...]
```

And I expected this to happen:

New composer.lock file is created but even existing one is emptied. composer install clears lock file as well.

**Comment by Seldaek** [comment]
Can you try again with `--no-plugins` just to make sure there is no third party interfering with Composer? I'm not sure what to say here.. this makes no sense to me and I cannot reproduce it..

**Comment by Terimor** [comment]
Thank you for your reply.

I'm not sure what this issue was related with, but I repeated my upgrade again, but gradually this time, package by package and I haven't encountered this issue anymore.
It also seemed that it was not related to content of composer.json, because I tried other 100% working files and this issue happened again.

Unfortunately I cannot provide any further details since I cannot reproduce it again and problem is solved.
